modules/options/src/main/scala/scala/build/options/BuildOptions.scala

```diff
@@ -33,6 +33,7 @@ final case class BuildOptions(
   scriptOptions: ScriptOptions = ScriptOptions(),
   internal: InternalOptions = InternalOptions(),
   mainClass: Option[String] = None,
+  mainClassLs: Option[Boolean] = None,
```

Comment:
I wouldn't put that option here. Listing main classes should only be a `cli` module concern, not a concern of the `build` module (which is kind of a library, so as much as possible, it shouldn't call `sys.exit` or basic `println` on its own).

Listing main classes could rely on `Build.Successful#foundMainClasses`, and be handled right after we get a successful build in commands that accept `--main-class-list`. Basically, instead of running / packaging / etc., we would print the main class list.

Reply:
should be fixed now